### PR TITLE
update to Redis overview and re-order structure [DOC-255]

### PR DESCRIPTION
**Pull request by harshini-rangaswamy**
# What changed, and why it matters

Updated overview content for Redis and restructured topics to follow the same style as Clickhouse and Flink

### CODE REVIEW

File: docs/products/redis/concepts/overview.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+* **Integration with data infrastructure:** Aiven ensures secure network connectivity using VPC peering, PrivateLink, or TransitGateway technologies. Aiven integrates with various observability tooling, including Datadog, Prometheus, and Jolokia, or you can use Aiven's observability tools for improved monitoring and logging.
+
+* **DevOps-friendly management and development:** Aiven for Redis is easy to manage using `Aiven Console <https://console.aiven.io/>`_, `Aiven CLI <https://github.com/aiven/aiven-client>`_, or :doc:`Terraform </docs/tools/terraform>` . Scaling, forking, and upgrading your Redis cluster is quick and easy. Aiven for Redis is compatible with open-source software, making it easy to use with your existing applications. Migrating between clouds and regions is also easy with Aiven for Redis.
+
```
Comment (hoang-vo) [inline docs/products/redis/concepts/overview.rst:27]: Shall we also add our Backup in this DevOps friendly section?

File: docs/products/redis/concepts/overview.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+Redis offers a wide range of data structures, including strings, hashes, lists, sets, sorted sets with range queries, bitmaps, hyperloglogs, geospatial indexes, and streams.
+
+Features and benefits of Aiven for Redis®*
+-------------------------------------------
+
+Aiven for Redis has many features that make it easy and stress-free to use:
+
+* **Managed service:** Aiven for Redis is fully managed, so you don't have to worry about setup, management, or updates. Aiven provides tools and integrations to help you easily use Redis in your data pipelines.
+
+* **Fast and easy deployment:** Aiven for Redis provides production-ready Redis clusters within a few minutes. You can deploy Redis to the cloud of your choice from 5 public clouds and over 100 regions. Aiven uses high-performance clusters with carefully selected instance types and storage options for top-notch performance. A Bring-your-own-account (BYOA) deployment model is available for strict control requirements.
```
Comment (hoang-vo) [inline docs/products/redis/concepts/overview.rst:22]: Shall we use Redis service instead? 

Redis Cluster is also the name of the plugin which RedisLab uses to support horizontal scaling. Tho w this context, it is understandable that this isnt "Redis Cluster".

File: docs/products/redis/concepts/overview.rst (hunk 1 above)
Comment (harshini-rangaswamy) [reply docs/products/redis/concepts/overview.rst:27]: Added this content as the next bullet list: 
**Backups and disaster recovery:** Aiven for Redis offers automatic and configurable backups. Automatic backups are taken every 24 hours, and the retention period varies based on the selected service plan.

File: docs/products/redis/concepts/overview.rst (hunk 2 above)
Comment (harshini-rangaswamy) [reply docs/products/redis/concepts/overview.rst]: Changed this to Redis service.

**Review by hoang-vo — APPROVED** [review]
I think it looks better now, nice work!

**Pull request merged**